sharedbuffers/mapped_struct.py

```diff
@@ -1408,15 +1408,37 @@ def unpack_from(cls, buf, offs, idmap = None):
         buf = _likerobuffer(buf)
         return cls(buf, offs, idmap)
 
+    @cython.locals(fast = cython.bint, dcode = cython.char, index = cython.longlong,
+        objlen = cython.longlong, dataoffs = cython.Py_ssize_t, itemsize = cython.uchar)
+    def getter(self, proxy_into = None, fast = False):
+        if proxy_into is None and fast:
```

Comment:
I think it would be worth it to delay constructing a generic proxy until you know that the datatype of the list is not purely numeric (i.e: dcode equal to 't' or 'T'). This way, you're eagerly creating a new object, even though in many cases it won't be needed.

Reply:
Good point